sympy/solvers/tests/test_solvers.py

```diff
@@ -114,6 +114,262 @@ def test_guess_transcendental():
     assert guess_solve_strategy(a*x**b - y, x)  # == GS_TRANSCENDENTAL
 
 
+def test_solve_io():
+    Y = 1 - y**2
+    X = 1 - x**2
+    flags = {}, dict(dict = True), dict(set = True)
+
+    ##### single one equation
+
+    d = {x: 0}
+    s = ([x], {(0,)})
+    syx = ([y, x], {(y, 0)})
+    sy = ([y], set())
+    vx = [0]
+    vyx = [(y, 0)]
+    e =                                                                         x
+    assert [solve(e, **f) for f in flags] ==            [vx,    [d],    s]      # no syms
+    assert [solve([e], **f) for f in flags] ==          [d,     [d],    s]
```

Comment:
These tests are horrible. I gave feedback that they should be written in a way that was more readable. The number one goal in readability of tests should be to make it so that the input and output are close together so that they can be compared.